internal/balancergroup/balancergroup_test.go

```diff
@@ -639,3 +754,84 @@ func (s) TestBalancerGracefulSwitch(t *testing.T) {
 		}
 	}
 }
+
+func (s) TestBalancerExitIdle_All(t *testing.T) {
+	balancerOne := "stub-balancer-test-balancer-group-exit-idle-one"
+	balancerTwo := "stub-balancer-test-balancer-group-exit-idle-two"
+	balancerThree := "stub-balancer-test-balancer-group-exit-idle-three"
+
+	balancerNames := []string{balancerOne, balancerTwo, balancerThree}
+	testIDs := []string{testBalancerIDs[0], testBalancerIDs[1], testBalancerIDs[2]}
+
+	exitIdleCh := make(chan string, len(balancerNames))
+
+	for _, name := range balancerNames {
+		stub.Register(name, stub.BalancerFuncs{
+			ExitIdle: func(_ *stub.BalancerData) {
+				exitIdleCh <- name
+			},
+		})
+	}
```

Comment:
I would get rid of the slices and the loops and instead keep this downright simple and dump.

```golang
  // Have two test IDs: testID1 and testID2
  // Have two channels, exitIdleCh1 and exitIdleCh2
  // Register two stub balancers, the first one for balancer1 that writes to exitIdleCh1 and the second one for balancer2 that writes to exitIdleCh2

  // Create the balancer group, and add the child balancers in there

  // Call ExitIdle

  // Validate that both exitIdleCh1 and exitIdleCh2 are written to. (I don't think it's an issue if one of those are written to more than once). In this case though, you might have to spawn goroutines to validate that both channels are written to, since we cannot guarantee the order in which they would be written to.
```